scalding-core/src/main/scala/com/twitter/scalding/DescribedExecutionApp.scala

```diff
@@ -0,0 +1,57 @@
+package com.twitter.scalding
+
+sealed abstract class ArgOptions
+
+object RequiredArg extends ArgOptions
+object OptionalArg extends ArgOptions
+object ListArg extends ArgOptions
+object BooleanArg extends ArgOptions
+
+case class DescribedArg(key: String, description: String, argOption: ArgOptions)
```

Comment:
Would using a builder style pattern not be cleaner/closer to what you had before?

DescribedArgs(args).withRequired("key"...).withRequired(...)